Client/SafeBrowsing/GSafeBrowsing.swift

```diff
@@ -0,0 +1,316 @@
+            completion(error)
+        }
+    }
+    
+    private func encode<T>(_ method: RequestType, endpoint: Endpoint, body: T) throws -> URLRequest where T: Encodable {
+        
+        let urlPath = "\(baseURL)\(endpoint.rawValue)?key=\(SafeBrowsingClient.apiKey)"
+        
+        guard let url = URL(string: urlPath) else {
+            throw SafeBrowsingError("Invalid Request")
+        }
+        
+        var request = URLRequest(url: url, cachePolicy: .reloadIgnoringLocalCacheData)
+        request.httpMethod = method.rawValue
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.setValue("application/json", forHTTPHeaderField: "Accept")
+        request.httpBody = try JSONEncoder().encode(body)
+        
+        return request
+    }
+    
+    @discardableResult
+    private func executeRequest<T>(_ request: URLRequest, type: T.Type, completion: @escaping (T?, Error?) -> Void) -> URLSessionDataTask where T: Decodable {
+        let task = session.dataTask(with: request) { data, response, error in
+            if let error = error {
+                return completion(nil, error)
+            }
+            
+            guard let data = data else {
+                return completion(nil, SafeBrowsingError("Invalid Server Response: No Data"))
+            }
+            
+            if let response = response as? HTTPURLResponse {
+                if response.statusCode < 200 || response.statusCode > 299 {
+                    do {
+                        let error = try JSONDecoder().decode(ResponseError.self, from: data)
+                        return completion(nil, SafeBrowsingError(error.message, code: error.code))
+                    } catch {
+                        return completion(nil, error)
+                    }
+                }
+            }
+            
+            do {
+                let response = try JSONDecoder().decode(type, from: data)
+                completion(response, nil)
+            } catch {
+                completion(nil, error)
+            }
+        }
+        task.resume()
+        return task
+    }
+    
+    private enum RequestType: String {
+        case get = "GET"
+        case post = "POST"
+    }
+    
+    private enum Endpoint: String {
+        case fetch = "/v4/threatListUpdates:fetch"
+        case fullHashes = "/v4/fullHashes:find"
+    }
+}
+
+extension URL {
+    func canonicalize() -> URL {
+        var absoluteString = self.absoluteString
+        absoluteString = absoluteString.replacingOccurrences(of: "\t", with: "")
+        absoluteString = absoluteString.replacingOccurrences(of: "\r", with: "")
+        absoluteString = absoluteString.replacingOccurrences(of: "\n", with: "")
+        
+        guard var components = URLComponents(string: absoluteString) else {
+            return self
+        }
+        
+        if var host = components.host?.removingPercentEncoding {
+            //TODO: Handle IP Addresses..
```

Comment:
Let's open an issue for this (and link to it from the comment) if it's not going to be part of this PR.